### DISCUSSION TITLE
behavior-mod-morphkey hotkey with three values

### DISCUSSION THREAD

**Discussion by antoniovmx** [nickcoutsos/keymap-editor#134] [Q&A] [answered]
Hey everyone,

I could use some more assistance with custom keys for the Kinesis Advantage 360 Pro keyboard. I want to set up a behavior-mod-morphkey with the following functions:

- Normal state: Outputs the number "2".

- Shift key pressed: Outputs a comma.

- Control key pressed: Outputs the greater than symbol (>).

Also, I have a quick question. Does anyone happen to know any keyboard shortcuts to select a single word, a phrase, and a paragraph using just the keyboard? Your insights would be greatly appreciated!

Thanks a bunch.

**Comment by nickcoutsos** [comment, marked as answer]
You should be able to achieve this by "chaining" mod-morph behaviors together. E.g., the modified version (let's say, when _shift_ is held) of one mod-morph does one thing, and anything else triggers a different mod-morph behavior. That latter mod-morph checks for the alternate modifier and triggers whichever character you'd wanted.

```dts
    shiftmod: shiftmod {
        compatible = "zmk,behavior-mod-morph";
        label = "SHIFTMOD";
        bindings =
            <&kp N2>,
            <&kp COMMA>;

        #binding-cells = <0>;
        mods = <(MOD_LSFT)>;
    };
    ctrlmod: ctrlmod {
        compatible = "zmk,behavior-mod-morph";
        label = "CTRLMOD";
        bindings =
            <&shiftmod>,
            <&kp GT>;

        #binding-cells = <0>;
        mods = <(MOD_LCTL)>;
    };
```

Then you'd bind `&ctrlmod` wherever you want to invoke this behavior. I'll leave it to you to name the behaviors more meaningfully.

Note that mod morphs don't accept parameters at bind-time so if you're doing this for all numbers you'd need to create the pair for each.

> Also, I have a quick question. Does anyone happen to know any keyboard shortcuts to select a single word, a phrase, and a paragraph using just the keyboard? Your insights would be greatly appreciated!

This may be platform and even program specific so that may be a challenge.

**Reply by antoniovmx** [reply]
It works perfectly. Thank you so much, Nick!
